### ISSUE TITLE
boesiger, 1974c listed as article in DROS5, but source points to a book chapter

### ISSUE THREAD

**Issue by jhpoelen** [TaxoDros/TaxoDros.github.io#19]
https://sandbox.zenodo.org/records/31753 uis a book chapter but in Bächlis' list it is a journal article? leave as is?

_Originally posted by @myrmoteras in https://github.com/bio-guoda/preston/issues/275#issuecomment-1969980923_

Associated DROS5 record retrieved via https://linker.bio/line:hash://md5/ff86b940567d278e50fa00672cf96629!/L17005-L17014 on 2024-02-29 - 

```
.TEXT;
boesiger, 1974c
.A Boesiger, E.,
.J 1974
.S The role of sexual selection in the maintenance
of the genetical heterogeneity of Drosophila
populations and its genetic basis.
.Z Front. Biol., 38:167-184.
.K ocr / part / p 813882 / 1974c
.P Boesiger, 1974c.pdf
```

indicates that the book chapter is archived as a article instead of a book. Expected a DROS5 record with 

```
.Z. Front. Biol., 38:167-184.
```

instead (note the period following (e.g., ```.Z.```))

**Comment by myrmoteras** [comment]
following our strategy to get Bächli data into Zenodo with minimal correction, besides data cleaning needed for successful upload, and the option for later improvements, I would go ahead

**Comment by jhpoelen** [comment]
@myrmoteras thanks for taking the time to review. The Zenodo records related to the TaxoDros literature can be traced by their LSIDs. These LSIDs can be used to update the Zenodo records with TaxoDros updates. 

Just to make sure - Would you like me to go ahead and upload all TaxoDros records to Zenodo (production not sandbox)?

**Comment by myrmoteras** [comment]
Yes, I would upload a couple to production and see what's happening
d

**Comment by jhpoelen** [comment]
Ok, I've uploaded two record - one existing one, and one new one. Please see #28 and #27 or https://zenodo.org/records/10728948 and https://zenodo.org/records/10728911

**Comment by myrmoteras** [comment]
Looks good.

Your idea to add "terrestrial" as keyword would be good.

In this https://zenodo.org/records/10728911, you added a new version. Did you do it because the files are different and because it has now its own DOI?

I add Terry, in case he has a moment to look at it. @Terry ***@***.***> these are the first two uploads to the BLR and taxodros and start of uploading all the rest. Any observations, comments or is it ok?


Cheers
donat

________________________________

**Comment by jhpoelen** [comment]
> In this https://zenodo.org/records/10728911, you added a new version. Did you do it because the files are different and because it has now its own DOI?

Not that fancy yet. The bot noticed an older version, and created a new one.

**Comment by myrmoteras** [comment]
How does the bot decide? What are the criteria?

**Comment by jhpoelen** [comment]
Each TaxoDros record has a lsid that marks a Zenodo record.

When updating a TaxoDros into Zenodo, the bot queries Zenodo for records with that lsid. If the lsid is not found, a new records is created, if a record with the lsid is found, a new version is created. If more than one record with the lsid are found, this is logged and no action is taken. 

 An example of such lsid is:
urn:lsid:taxodros.uzh.ch:id:abd%20el-halim%20et%20al.%2C%202005

with associated record: 

```
.TEXT;
abd el-halim et al., 2005
.A Abd El-Halim, A.S., Mostafa, A.A.,
& Allam, K.A.M.a.,
.J 2005
.S Dipterous flies species and their densities in
fourteen Egyptian governorates.
.Z J. Egypt. Soc. Parasitol., 35:351-362.
.K ocr
.P Abd El-Halim et al., 2005.pdf
```

Note how the lsid is derived from the id that Bächli used in the DROS5 records (i.e., 'abd el-halim et al., 2005').

So, the lsid builds a bridge from the Zenodo universe to the TaxoDros universe.

I hope this answers your question.

**Comment by myrmoteras** [comment]
Ok
tx

**Comment by jhpoelen** [comment]
@myrmoteras would you like me to proceed uploading the entire corpus. Or would you rather wait until others have reviewed the test records. Please advise.

**Comment by jhpoelen** [comment]
The bot only touches records with taxodros lsids. Assuming that treatmentbank does not include TaxoDros lsids, the 160 fruit fly pubs in Treatment Bank will not be messed with by the bot.

**Comment by jhpoelen** [comment]
I see treatment bank and the TaxoDros corpus as separate corpora. Their records may be linked at some point, and this requires some curatorial decisions/workflows/work to the put in place.

**Comment by myrmoteras** [comment]
But then we create duplicates which is against the rule.
d

**Comment by myrmoteras** [comment]
> I see treatment bank and the TaxoDros corpus as separate corpora. Their records may be linked at some point, and this requires some curatorial decisions/workflows/work to the put in place.

then we need to get the TaxoDros lsids into the existing records.

**Comment by jhpoelen** [comment]
Perhaps best to have a video chat on how to proceed. Let me know your availability.

**Comment by myrmoteras** [comment]
Now?

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by jhpoelen** [comment]
ok, https://whereby.com/jorritpoelen

**Comment by jhpoelen** [comment]
@myrmoteras thanks for the chat. As discussed, I'll schedule a first round of uploading the TaxoDros corpus.
